lib/property-exists.js

```diff
@@ -0,0 +1,13 @@
+module.exports = function checkNestedProperty (obj, pathString) {
+  var path = pathString.split('.');
+
+  var property = path.reduce((prev, curr) => {
+    if (prev[curr]) {
+      return prev[curr];
+    } else {
+      return false;
+    }
+  }, obj);
```

Comment:
In this case it's better to use a for loop to terminate early, as you have it now we have to traverse the whole `pathString`, and we might end up checking for `(false)['x']`, which could result in a false positive if, for instance the property name was 'constructor'